core/ledger/kvledger/txmgmt/txmgr/lockbasedtxmgr/txmgr_test.go

```diff
@@ -31,6 +31,7 @@ func TestTxSimulatorWithNoExistingData(t *testing.T) {
 		t.Logf("Running test for TestEnv = %s", testEnv.getName())
 		testLedgerID := "testtxsimulatorwithnoexistingdata"
 		testEnv.init(t, testLedgerID, nil)
+		defer testEnv.cleanup()
```

Comment:
in general, `defer` in loop often leads to unexpected behavior and should be used with caution - in this case, you will be calling `cleanup` on the **last** `testEnv` multiple times

Either clean up w/o defer (like what we already have here), or assign testEnv to a new var and call defer.